cluster/gce/util.sh

```diff
@@ -843,6 +843,11 @@ function construct-windows-kubelet-flags {
     flags+=" --node-labels=${node_labels}"
   fi
 
+  # Apply Windows-only node taints.
+  if [[ -n "${WINDOWS_NODE_TAINTS:-}" ]]; then
+    flags+=" --register-with-taints=${WINDOWS_NODE_TAINTS}"
```

Comment:
At line 758 we're already setting --register-with-taints with the common (Linux and Windows) NODE_TAINTS. Can you move this flag out of construct-common-kubelet-flags and set it correctly here and in construct-linux-kubelet-flags?

Even if setting the flag twice does what we want (applies both sets of node taints), it's confusing.